python/paddle/fluid/transpiler/distribute_transpiler.py

```diff
@@ -540,17 +626,19 @@ def _get_splited_name_and_shape(varname):
 
         # 2. rename op outputs
         for op in orig_s_prog.global_block().ops:
+
             new_outputs = dict()
             # do not append startup op if var is not on this pserver
             op_on_pserver = False
-            for key in op.output_names:
-                newname, _ = _get_splited_name_and_shape(op.output(key)[0])
-                if newname:
-                    op_on_pserver = True
-                    new_outputs[key] = created_var_map[newname]
-                elif op.output(key)[0] in pserver_vars:
-                    op_on_pserver = True
-                    new_outputs[key] = pserver_vars[op.output(key)[0]]
+            if op.type not in ["recv", "fetch_barrier", "concat"]:
```

Comment:
Do we have a way to avoid the hard code op name, such as using `OpRole`.

Reply:
Maybe tagging is a good way, but we have not tagging yet..

Reply:
That's okay, maybe you can add a TODO comment here.

Reply:
Cool idea! thanks! I will add it later